### PR TITLE
gomarkdoc: init at 1.1.0

### PR DESCRIPTION
**Pull request by brpaz**
## Description of changes

Add [gomarkdoc](https://github.com/princjef/gomarkdoc), a package to Generate markdown documentation for Golang code. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2744,6 +2744,12 @@
     githubId = 40476330;
     name = "brokenpip3";
   };
+  brpaz = {
+    email = "oss@brunopaz.dev";
+    github = "brpaz";
+    githubId = 184563;
+    name = "Bruno Paz";
+  };
```
Comment (Janik-Haag) [inline maintainers/maintainer-list.nix:2752]: this should be it's own commit called `maintainers: add brpaz`

File: pkgs/development/tools/gomarkdoc/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{ lib, buildGoModule, fetchFromGitHub, testers, gomarkdoc }:
+
+buildGoModule rec {
+  pname = "gomarkdoc";
+  version = "1.1.0";
+  src = fetchFromGitHub {
```
Comment (Janik-Haag) [inline pkgs/development/tools/gomarkdoc/default.nix:6]:
```suggestion
  version = "1.1.0";

  src = fetchFromGitHub {
```
small formatting thing

File: pkgs/development/tools/gomarkdoc/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+   passthru.tests = {
+    version = testers.testVersion {
+      package = gomarkdoc;
+    };
+  };
+
+  meta = with lib; {
+    description = "Generate markdown documentation for Go (golang) code";
+    homepage = "https://github.com/princjef/gomarkdoc";
+
```
Comment (Janik-Haag) [inline pkgs/development/tools/gomarkdoc/default.nix:33]:
```suggestion
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -28404,6 +28404,8 @@ with pkgs;
 
   goimports-reviser = callPackage ../development/tools/goimports-reviser { };
 
+  gomarkdoc = callPackage ../development/tools/gomarkdoc {};
```
Comment (Janik-Haag) [inline pkgs/top-level/all-packages.nix:28407]:
```suggestion
  gomarkdoc = callPackage ../development/tools/gomarkdoc { };
````
formatting nit. 

you don't have to do this now but in the future you can just put the package into `pkgs/by-name` and it will be picked up automatically

File: pkgs/development/tools/gomarkdoc/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
+{ lib, buildGoModule, fetchFromGitHub, testers, gomarkdoc }:
```
Comment (Janik-Haag) [inline pkgs/development/tools/gomarkdoc/default.nix:1]:
```suggestion
{ lib
, buildGoModule
, fetchFromGitHub
, testers
, gomarkdoc
}:
```
makes future diffs easier to read

File: pkgs/development/tools/gomarkdoc/default.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+buildGoModule rec {
+  pname = "gomarkdoc";
+  version = "1.1.0";
+  src = fetchFromGitHub {
+    owner = "princjef";
+    repo = "gomarkdoc";
+    rev = "v${version}";
+    sha256 = "sha256-eMH+F1ZXAKHqnrvOJvCETm2NiDwY03IFHrDNYr3jaW8=";
```
Comment (Janik-Haag) [inline pkgs/development/tools/gomarkdoc/default.nix:10]:
```suggestion
    hash = "sha256-eMH+F1ZXAKHqnrvOJvCETm2NiDwY03IFHrDNYr3jaW8=";
```
the sha256 key is more or less deprecated and you should use hash instead.

**Review by Janik-Haag — COMMENTED** [review]
Hi, welcome to nixpkgs! 
We try to have a atomic git history, so this pr should have exactly two commits, one for adding your self as maintainer, and one for adding the package.
called 
`maintainers: add brpaz`
and
`gomarkdoc: init at 1.1.0`
respectively

**Comment by brpaz** [comment]
@Janik-Haag I think I did all the requested changes. Please review.

Since the GitHub Actions were failing, I also moved the package to `pkg/by-name` as suggested.

**Review by Janik-Haag — APPROVED** [review]
LGTM
